pyproject.toml

```diff
@@ -25,20 +25,11 @@ classifiers = [
     "Topic :: Scientific/Engineering",
 ]
 requires-python = ">=3.9"
-dependencies = [
-    "pvlib>=0.9.0",
-    "shapely>=2.0",
-    "matplotlib",
-    "numpy",
-    "pandas",
-]
+dependencies = ["pvlib>=0.9.0", "shapely>=2.0", "numpy", "pandas"]
 
 [project.optional-dependencies]
-test = [
-    "pytest>=3.2.1",
-    "pytest-mock>=1.10.0",
-    "mock",
-]
+plot = ["matplotlib"]
+test = ["pytest>=3.2.1", "pytest-mock>=1.10.0", "mock", "matplotlib"]
 doc = [
```

Comment:
Please also add matplotlib to `doc`.  It happens to gets pulled in anyway by `ipykernel`, but better to require it directly.